### ISSUE TITLE
Issue when using babel.

### ISSUE THREAD

**Issue by Hypercubed** [josdejong/typed-function#9]
For some reason when using babel unnamed functions get a function name.  You can see this in action [here](https://babeljs.io/repl/#?experimental=true&evaluate=false&loose=true&spec=true&code=%0Avar%20obj%20%3D%20%7B%0A%20%20'number'%3A%20function%20%28a%29%20%7B%0A%20%20%20%20return%20'a%20is%20a%20number'%3B%0A%20%20%7D%2C%0A%20%20'number%2C%20boolean'%3A%20function%20%28a%2C%20b%29%20%7B%0A%20%20%20%20return%20'a%20is%20a%20number%2C%20b%20is%20a%20boolean'%3B%0A%20%20%7D%2C%0A%20%20'number%2C%20number'%3A%20function%20%28a%2C%20b%29%20%7B%0A%20%20%20%20return%20'a%20is%20a%20number%2C%20b%20is%20a%20number'%3B%0A%20%20%7D%0A%7D%3B%0A%0Avar%20b%20%3D%20function%20%28a%29%20%7B%0A%20%20return%20'a%20is%20a%20number'%3B%0A%7D).  This causes typed-function to fail the name check in https://github.com/josdejong/typed-function/blob/master/typed-function.js#L76 .

I guess a user can include matching names their code but any dependencies that run through babel may fail.  Perhaps disable this check and use the first name by default?

I imagine this could also cause trouble with minifiers that may strip function names, but I haven't verified that.

**Comment by josdejong** [comment]
Thanks for bringing this up. That means that it's not reliable to use `fn.name`. So instead we could either skip using .name completely, or add a different "name" property to functions with the name set by typed-function, could be something like `fn.typedName`.

**Comment by Hypercubed** [comment]
How about just a `name` property on the config object that overide the name check?  I can do a PR...  Although, I guess I don't really understand the need for the name at all.

**Comment by Hypercubed** [comment]
I just realized we can already do this with typed(String, Object).  I guess the issue may remain for other code that may go through the transpiler or minifier.

**Comment by josdejong** [comment]
yes indeed, this stays problematic with minification/transpiliation so it should be solved.

**Comment by yongjun21** [comment]
Encountered the same issue on the latest release of Chrome. Throwing

```
SyntaxError: Unexpected token |
```

whenever a union type is passed in. Seems the anonymous functions got assigned the method names.

Urgently need a fix as it is now affecting not just minification/transpilation but many typed-function based libraries required by codes running on Chrome.

**Comment by josdejong** [comment]
Thanks for reporting @yongjun21 . How can I reproduce this issue?

**Comment by yongjun21** [comment]
@josdejong  on http://jsbin.com/hutucuzoni/edit?html,output

Add a line:

```
print(math.median(math.ones(5)));
```

**Comment by josdejong** [comment]
Thanks, that jsbin indeed fails. 

However when I update to the latest version of math.js (v3.2.1) instead of v3.1.4, the issue is resolved (see https://github.com/josdejong/mathjs/issues/601, applied in [v3.2.0](http://mathjs.org/history.html#20160416-version-320)). 

I've updated http://jsbin.com/devacu/edit?html,output to the latest version too now.

I think we just forgot to close this `typed-function` issue #9, unless I'm overlooking something it has been solved already.
